### DISCUSSION TITLE
MLP in Allegro

### DISCUSSION THREAD

**Discussion by kudoryutaro** [mir-group/allegro#47] [Q&A]
Why does allegro's MLP use fx.Graph() and fx.proxy.GraphAppendingTracer(graph)?
Any reason for using fx.Graph() instead of the usual torch.nn.Linear?
Is it faster to use fx.Graph()?

**Comment by Linux-cpp-lisp** [comment]
Hi @kudoryutaro ,

`torch.fx` is an API for building `torch.nn.Module`s that can use any PyTorch operations to do the computations: https://pytorch.org/docs/stable/fx.html.  In practice, I don't think there is any user-relevant difference between our MLP implementation and the more basic one you are describing.
